### PR TITLE
SECURITY: ensures timeouts are correctly used on connect

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: lib/oneboxer.rb [hunk 1]
```diff
@@ -407,7 +407,8 @@ def self.external_onebox(url, available_strategies = nil)
         ignore_hostnames: blocked_domains,
         force_get_hosts: force_get_hosts,
         force_custom_user_agent_hosts: force_custom_user_agent_hosts,
-        preserve_fragment_url_hosts: preserve_fragment_url_hosts
+        preserve_fragment_url_hosts: preserve_fragment_url_hosts,
+        timeout: 5
```
Comment (jjaffeux) [inline lib/oneboxer.rb:411]: We were using default FD timeout, which is 20 seconds and might be slightly too much?

File: lib/onebox/helpers.rb [hunk 2]
```diff
@@ -118,8 +117,7 @@ def self.fetch_response(location, redirect_limit: 5, domain: nil, headers: nil,
     def self.fetch_content_length(location)
       uri = URI(location)
 
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.is_a?(URI::HTTPS)) do |http|
-        http.open_timeout = Onebox.options.connect_timeout
+      Net::HTTP.start(uri.host, uri.port, open_timeout: Onebox.options.connect_timeout, use_ssl: uri.is_a?(URI::HTTPS)) do |http|
```
Comment (jjaffeux) [inline lib/onebox/helpers.rb:120]: start will attempt to connect right away, setting the open_timeout in the block is too late

File: lib/onebox/helpers.rb [hunk 3]
```diff
@@ -63,8 +63,7 @@ def self.fetch_response(location, redirect_limit: 5, domain: nil, headers: nil,
       end
 
       result = StringIO.new
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.normalized_scheme == 'https') do |http|
-        http.open_timeout = Onebox.options.connect_timeout
+      Net::HTTP.start(uri.host, uri.port, open_timeout: Onebox.options.connect_timeout, use_ssl: uri.normalized_scheme == 'https') do |http|
```
Comment (jjaffeux) [inline lib/onebox/helpers.rb:66]: start will attempt to connect right away, setting the open_timeout in the block is too late

File: lib/final_destination.rb [hunk 4]
```diff
@@ -196,6 +196,7 @@ def resolve
     response = Excon.public_send(@http_verb,
       @uri.to_s,
       read_timeout: timeout,
+      connect_timeout: timeout,
```
Comment (jjaffeux) [inline lib/final_destination.rb:199]: we were using default excon connect_timeout which is too high

**Review by tgxworld — APPROVED** [review]

**Review by jbrw — APPROVED** [review]

**Pull request merged**